src/Xeno/SAX.hs

```diff
@@ -183,8 +213,8 @@ process openF attrF endOpenF textF closeF str = findLT 0
                                                  (quoteIndex + 1)
                                                  (endQuoteIndex))
                                             findAttributes (endQuoteIndex + 1)
-                                   else throw XenoParseError
-                         else throw XenoParseError
+                                   else throw (XenoParseError ("Expected ' or \", got: " <> S.singleton usedChar))
+                         else throw (XenoParseError ("Expected =, got: " <> S.singleton (s_index str afterAttrName) <> " at character index: " <> (S8.pack . show) afterAttrName))
```

Comment:
These two computations may add a speed hit because GHC can't immediately throw everything away before throwing the exception, it uses the variables in scope. So in a sense it creates more work for the GHC runtime. I recall including something like this and it adding some microseconds or something.

Anyway, we'll see what the overall benchmark difference is.